lib/Sources/BluetoothNative/Options+IncludeAdvertisementEvent.swift

```diff
@@ -0,0 +1,118 @@
+import Bluetooth
+import BluetoothClient
+import Foundation
+
+extension Options {
+    func includeAdvertisementEventInDeviceList(_ advertisementEvent: AdvertisementEvent) -> Bool {
+        guard acceptAllDevices == false else {
+            return true
+        }
+
+        let matchesOnFilters = self.filters?.compactMap { $0.matches(with: advertisementEvent) }.contains { $0 == true } ?? true
+
+        let matchesOnExclusionFilters = self.exclusionFilters?.compactMap { $0.matches(with: advertisementEvent) }.contains { $0 == true } ?? false
```

Comment:
The `Options.Filter.matches` can be simplified slightly if you extract the `Advertisement` as a variable:

```suggestion
        let advertisement = advertisementEvent.advertisement
        let matchesOnFilters = self.filters?.contains { $0.matches(with: advertisement) } ?? true
        let matchesOnExclusionFilters = self.exclusionFilters?.contains { $0.matches(with: advertisement) } ?? false
```

Reply:
Oh yeah, nice.